WordPress/gutenberg
# Editor: Unify the DocumentTools component

Related #52632 

## What?

This PR continues the work on the great unification between post and site editors. This PR unifies the DocumentTools (left part of the editor header toolbar) component between edit site and edit post.

## Testing instructions

- Open both post and site editors and ensure the header left toolbar (inserter, undo, redo, tool selection, list view) buttons work as intended.
